<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>precost_call</class>
 <widget class="QDialog" name="precost_call">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>922</width>
    <height>706</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <pointsize>11</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <widget class="QLineEdit" name="year_widget">
   <property name="geometry">
    <rect>
     <x>230</x>
     <y>36</y>
     <width>51</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
  </widget>
  <widget class="QLabel" name="user">
   <property name="geometry">
    <rect>
     <x>745</x>
     <y>5</y>
     <width>51</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>사용자 :</string>
   </property>
  </widget>
  <widget class="Line" name="line_6">
   <property name="geometry">
    <rect>
     <x>150</x>
     <y>74</y>
     <width>20</width>
     <height>61</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
    </font>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="Line" name="line">
   <property name="geometry">
    <rect>
     <x>12</x>
     <y>17</y>
     <width>891</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
    </font>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="Line" name="line_13">
   <property name="geometry">
    <rect>
     <x>6</x>
     <y>238</y>
     <width>901</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
    </font>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QLabel" name="hang_1">
   <property name="geometry">
    <rect>
     <x>60</x>
     <y>79</y>
     <width>21</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>항</string>
   </property>
  </widget>
  <widget class="QLabel" name="hun_Y">
   <property name="geometry">
    <rect>
     <x>196</x>
     <y>40</y>
     <width>36</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>년도 :</string>
   </property>
  </widget>
  <widget class="Line" name="line_4">
   <property name="geometry">
    <rect>
     <x>12</x>
     <y>160</y>
     <width>891</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
    </font>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QLabel" name="marks_1">
   <property name="geometry">
    <rect>
     <x>760</x>
     <y>146</y>
     <width>56</width>
     <height>12</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>비고</string>
   </property>
  </widget>
  <widget class="QLabel" name="amount_1">
   <property name="geometry">
    <rect>
     <x>410</x>
     <y>144</y>
     <width>51</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>금액</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="cost_date">
   <property name="geometry">
    <rect>
     <x>13</x>
     <y>40</y>
     <width>63</width>
     <height>14</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>지출일자 :</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="marks_widget">
   <property name="geometry">
    <rect>
     <x>651</x>
     <y>170</y>
     <width>251</width>
     <height>30</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
    </font>
   </property>
  </widget>
  <widget class="Line" name="line_2">
   <property name="geometry">
    <rect>
     <x>12</x>
     <y>64</y>
     <width>891</width>
     <height>16</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="Line" name="line_22">
   <property name="geometry">
    <rect>
     <x>2</x>
     <y>685</y>
     <width>901</width>
     <height>20</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QLabel" name="user_widget">
   <property name="geometry">
    <rect>
     <x>800</x>
     <y>3</y>
     <width>56</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QPushButton" name="add_cost_button">
   <property name="geometry">
    <rect>
     <x>60</x>
     <y>208</y>
     <width>141</width>
     <height>34</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>13</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>변동사항 등록</string>
   </property>
  </widget>
  <widget class="Line" name="line_5">
   <property name="geometry">
    <rect>
     <x>12</x>
     <y>195</y>
     <width>891</width>
     <height>16</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QLineEdit" name="special_account_detail_widget">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>169</y>
     <width>331</width>
     <height>30</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
    </font>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="alignment">
    <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
   </property>
  </widget>
  <widget class="QPushButton" name="cost_cancel_button">
   <property name="geometry">
    <rect>
     <x>807</x>
     <y>33</y>
     <width>91</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>12</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="focusPolicy">
    <enum>Qt::NoFocus</enum>
   </property>
   <property name="text">
    <string>닫기(취소)</string>
   </property>
  </widget>
  <widget class="QPushButton" name="cost_save_button">
   <property name="geometry">
    <rect>
     <x>690</x>
     <y>33</y>
     <width>101</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>12</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="focusPolicy">
    <enum>Qt::NoFocus</enum>
   </property>
   <property name="text">
    <string>저장(F12)</string>
   </property>
   <property name="shortcut">
    <string>F12</string>
   </property>
  </widget>
  <widget class="QLabel" name="hap_total">
   <property name="geometry">
    <rect>
     <x>250</x>
     <y>218</y>
     <width>71</width>
     <height>21</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>등록 총액</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="cost_hap_widget">
   <property name="geometry">
    <rect>
     <x>323</x>
     <y>216</y>
     <width>113</width>
     <height>26</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="focusPolicy">
    <enum>Qt::NoFocus</enum>
   </property>
   <property name="alignment">
    <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
   </property>
  </widget>
  <widget class="QDateEdit" name="cost_date_widget">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>35</y>
     <width>111</width>
     <height>22</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="focusPolicy">
    <enum>Qt::NoFocus</enum>
   </property>
   <property name="inputMethodHints">
    <set>Qt::ImhDate|Qt::ImhPreferNumbers</set>
   </property>
   <property name="correctionMode">
    <enum>QAbstractSpinBox::CorrectToNearestValue</enum>
   </property>
   <property name="currentSection">
    <enum>QDateTimeEdit::DaySection</enum>
   </property>
   <property name="date">
    <date>
     <year>2022</year>
     <month>1</month>
     <day>1</day>
    </date>
   </property>
  </widget>
  <widget class="QLabel" name="hun_month">
   <property name="geometry">
    <rect>
     <x>290</x>
     <y>37</y>
     <width>21</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>월</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="month_widget">
   <property name="geometry">
    <rect>
     <x>310</x>
     <y>36</y>
     <width>31</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
  </widget>
  <widget class="QLabel" name="hun_week">
   <property name="geometry">
    <rect>
     <x>360</x>
     <y>36</y>
     <width>71</width>
     <height>21</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>몇번째 주 :</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="week_widget">
   <property name="geometry">
    <rect>
     <x>440</x>
     <y>36</y>
     <width>31</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
  </widget>
  <widget class="QLabel" name="cost_detail_1">
   <property name="geometry">
    <rect>
     <x>120</x>
     <y>145</y>
     <width>51</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>적요</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="mok_1">
   <property name="geometry">
    <rect>
     <x>212</x>
     <y>80</y>
     <width>21</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>목</string>
   </property>
  </widget>
  <widget class="QLabel" name="semok_1">
   <property name="geometry">
    <rect>
     <x>400</x>
     <y>79</y>
     <width>31</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>세목</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="amount_widget">
   <property name="geometry">
    <rect>
     <x>369</x>
     <y>169</y>
     <width>131</width>
     <height>30</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
    </font>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="alignment">
    <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
   </property>
  </widget>
  <widget class="QComboBox" name="hang_combo_widget">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>104</y>
     <width>141</width>
     <height>28</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
    </font>
   </property>
  </widget>
  <widget class="QComboBox" name="mok_combo_widget">
   <property name="geometry">
    <rect>
     <x>166</x>
     <y>103</y>
     <width>151</width>
     <height>30</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
    </font>
   </property>
  </widget>
  <widget class="QComboBox" name="semok_combo_widget">
   <property name="geometry">
    <rect>
     <x>335</x>
     <y>101</y>
     <width>171</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
    </font>
   </property>
  </widget>
  <widget class="QTableWidget" name="changed_detail_tableWidget">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>250</y>
     <width>891</width>
     <height>441</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="focusPolicy">
    <enum>Qt::NoFocus</enum>
   </property>
   <property name="sizeAdjustPolicy">
    <enum>QAbstractScrollArea::AdjustToContents</enum>
   </property>
   <column>
    <property name="text">
     <string>항</string>
    </property>
   </column>
   <column>
    <property name="text">
     <string>목</string>
    </property>
   </column>
   <column>
    <property name="text">
     <string>세목</string>
    </property>
   </column>
   <column>
    <property name="text">
     <string>적요</string>
    </property>
   </column>
   <column>
    <property name="text">
     <string>금액</string>
    </property>
   </column>
   <column>
    <property name="text">
     <string>입출계좌</string>
    </property>
   </column>
   <column>
    <property name="text">
     <string>증감</string>
    </property>
   </column>
   <column>
    <property name="text">
     <string>비고</string>
    </property>
   </column>
  </widget>
  <widget class="QPushButton" name="hap_recalculator_button">
   <property name="geometry">
    <rect>
     <x>480</x>
     <y>218</y>
     <width>101</width>
     <height>23</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="focusPolicy">
    <enum>Qt::NoFocus</enum>
   </property>
   <property name="text">
    <string>합계 재계산</string>
   </property>
  </widget>
  <widget class="QLabel" name="bank_in_out_label">
   <property name="geometry">
    <rect>
     <x>529</x>
     <y>144</y>
     <width>91</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>입출금계좌</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QComboBox" name="banks_combo_widget">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>170</y>
     <width>111</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
    </font>
   </property>
  </widget>
  <widget class="QPushButton" name="remove_row_Button">
   <property name="geometry">
    <rect>
     <x>660</x>
     <y>211</y>
     <width>101</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="focusPolicy">
    <enum>Qt::NoFocus</enum>
   </property>
   <property name="text">
    <string>선택행 삭제</string>
   </property>
  </widget>
  <widget class="QPushButton" name="cost_serch_button">
   <property name="geometry">
    <rect>
     <x>537</x>
     <y>35</y>
     <width>131</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>12</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="focusPolicy">
    <enum>Qt::NoFocus</enum>
   </property>
   <property name="text">
    <string>이전 내역 검색</string>
   </property>
  </widget>
  <widget class="QPushButton" name="cost_reset_button">
   <property name="geometry">
    <rect>
     <x>780</x>
     <y>210</y>
     <width>121</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="focusPolicy">
    <enum>Qt::NoFocus</enum>
   </property>
   <property name="text">
    <string>내역 초기화</string>
   </property>
  </widget>
  <widget class="Line" name="line_7">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>93</y>
     <width>891</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
    </font>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="Line" name="line_9">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>128</y>
     <width>891</width>
     <height>16</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="Line" name="line_8">
   <property name="geometry">
    <rect>
     <x>317</x>
     <y>72</y>
     <width>20</width>
     <height>61</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
    </font>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="Line" name="line_10">
   <property name="geometry">
    <rect>
     <x>350</x>
     <y>140</y>
     <width>20</width>
     <height>61</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
    </font>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="Line" name="line_11">
   <property name="geometry">
    <rect>
     <x>500</x>
     <y>139</y>
     <width>20</width>
     <height>61</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
    </font>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="Line" name="line_12">
   <property name="geometry">
    <rect>
     <x>630</x>
     <y>138</y>
     <width>20</width>
     <height>61</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
    </font>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="QComboBox" name="balance_combo_widget">
   <property name="geometry">
    <rect>
     <x>634</x>
     <y>103</y>
     <width>151</width>
     <height>30</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
  </widget>
  <widget class="QLabel" name="in_de_label">
   <property name="geometry">
    <rect>
     <x>550</x>
     <y>107</y>
     <width>71</width>
     <height>21</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>11</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>증가 감소</string>
   </property>
  </widget>
 </widget>
 <tabstops>
  <tabstop>year_widget</tabstop>
  <tabstop>month_widget</tabstop>
  <tabstop>week_widget</tabstop>
  <tabstop>hang_combo_widget</tabstop>
  <tabstop>mok_combo_widget</tabstop>
  <tabstop>semok_combo_widget</tabstop>
  <tabstop>balance_combo_widget</tabstop>
  <tabstop>special_account_detail_widget</tabstop>
  <tabstop>amount_widget</tabstop>
  <tabstop>banks_combo_widget</tabstop>
  <tabstop>marks_widget</tabstop>
  <tabstop>add_cost_button</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>add_cost_button</sender>
   <signal>clicked()</signal>
   <receiver>precost_call</receiver>
   <slot>data_append()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>200</x>
     <y>168</y>
    </hint>
    <hint type="destinationlabel">
     <x>213</x>
     <y>152</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>remove_row_Button</sender>
   <signal>clicked()</signal>
   <receiver>precost_call</receiver>
   <slot>remove_row()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>691</x>
     <y>158</y>
    </hint>
    <hint type="destinationlabel">
     <x>635</x>
     <y>154</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hap_recalculator_button</sender>
   <signal>clicked()</signal>
   <receiver>precost_call</receiver>
   <slot>re_calculate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>489</x>
     <y>157</y>
    </hint>
    <hint type="destinationlabel">
     <x>472</x>
     <y>144</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cost_serch_button</sender>
   <signal>clicked()</signal>
   <receiver>precost_call</receiver>
   <slot>serch_cost()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>567</x>
     <y>56</y>
    </hint>
    <hint type="destinationlabel">
     <x>532</x>
     <y>46</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cost_save_button</sender>
   <signal>clicked()</signal>
   <receiver>precost_call</receiver>
   <slot>file_save()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>709</x>
     <y>44</y>
    </hint>
    <hint type="destinationlabel">
     <x>668</x>
     <y>47</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cost_cancel_button</sender>
   <signal>clicked()</signal>
   <receiver>precost_call</receiver>
   <slot>file_close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>826</x>
     <y>44</y>
    </hint>
    <hint type="destinationlabel">
     <x>800</x>
     <y>41</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>data_append()</slot>
  <slot>file_save()</slot>
  <slot>re_calculate()</slot>
  <slot>file_close()</slot>
  <slot>remove_row()</slot>
 </slots>
</ui>
